frontend/pkg/frontend/frontend.go

```diff
@@ -296,17 +323,55 @@ func (f *Frontend) ArmResourceCreateOrUpdate(writer http.ResponseWriter, request
 
 	// URL path is already lowercased by middleware.
 	resourceID := request.URL.Path
+	resourceGroup := request.PathValue(PathSegmentResourceGroupName)
 	subscriptionID := request.PathValue(PathSegmentSubscriptionID)
 
-	cluster, updating := f.cache.GetCluster(resourceID)
-	versionedCurrentCluster := versionedInterface.NewHCPOpenShiftCluster(cluster)
+	var doc *database.HCPOpenShiftClusterDocument
+	var updating bool = true
+	doc, err = f.dbClient.GetClusterDoc(ctx, resourceID, subscriptionID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
```

Comment:
Similar comment as above -- maybe we want to ask Cluster Service to search to make sure the cluster _really_ doesn't exist before initiating a cluster creation.